File: src/test/java/no/novari/flyt/archive/gateway/dispatch/mapping/DokumentbeskrivelseMappingServiceTest.java
package no.novari.flyt.archive.gateway.dispatch.mapping;

import no.fint.model.resource.Link;
import no.fint.model.resource.arkiv.noark.DokumentbeskrivelseResource;
import no.novari.flyt.archive.gateway.dispatch.model.instance.DokumentbeskrivelseDto;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.util.Collections;
import java.util.HashMap;
import java.util.UUID;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(MockitoExtension.class)
class DokumentbeskrivelseMappingServiceTest {

    @InjectMocks
    private DokumentbeskrivelseMappingService dokumentbeskrivelseMappingService;

    @SuppressWarnings("unused")
    @Mock
    private SkjermingMappingService skjermingMappingService;

    @Test
    void toDokumentbeskrivelseResource() {
        DokumentbeskrivelseDto dto = DokumentbeskrivelseDto.builder()
                .tittel("testTittel")
                .dokumentstatus("testDokumentstatus")
                .dokumentType("testDokumenttype")
                .tilknyttetRegistreringSom("testTilknyttetRegistreringSom")
                .build();

        HashMap<UUID, Link> linkMap = new HashMap<>();

        DokumentbeskrivelseResource resource = dokumentbeskrivelseMappingService.toDokumentbeskrivelseResource(dto, linkMap);

        assertThat(resource.getTittel()).isEqualTo("testTittel");
        assertThat(resource.getDokumentType()).isEqualTo(Collections.singletonList(Link.with("testDokumenttype")));
        assertThat(resource.getDokumentstatus()).isEqualTo(Collections.singletonList(Link.with("testDokumentstatus")));
        assertThat(resource.getTilknyttetRegistreringSom()).isEqualTo(Collections.singletonList(Link.with("testTilknyttetRegistreringSom")));
    }
}

File: src/test/kotlin/no/novari/flyt/archive/gateway/dispatch/mapping/DokumentbeskrivelseMappingServiceTest.kt
package no.novari.flyt.archive.gateway.dispatch.mapping

import no.novari.fint.model.resource.Link
import no.novari.flyt.archive.gateway.dispatch.model.instance.DokumentbeskrivelseDto
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DokumentbeskrivelseMappingServiceTest {
    private val dokumentbeskrivelseMappingService =
        DokumentbeskrivelseMappingService(
            DokumentObjektMappingService(),
            SkjermingMappingService(),
        )

    @Test
    fun `maps to DokumentbeskrivelseResource`() {
        val dto =
            DokumentbeskrivelseDto
                .builder()
                .tittel("testTittel")
                .dokumentstatus("testDokumentstatus")
                .dokumentType("testDokumenttype")
                .tilknyttetRegistreringSom("testTilknyttetRegistreringSom")
                .build()

        val resource = dokumentbeskrivelseMappingService.toDokumentbeskrivelseResource(dto, emptyMap())

        assertThat(resource.tittel).isEqualTo("testTittel")
        assertThat(resource.dokumentType).isEqualTo(listOf(Link.with("testDokumenttype")))
        assertThat(resource.dokumentstatus).isEqualTo(listOf(Link.with("testDokumentstatus")))
        assertThat(resource.tilknyttetRegistreringSom).isEqualTo(listOf(Link.with("testTilknyttetRegistreringSom")))
    }
}
